### PR TITLE
added overflow limits to some text fields and fixed some overflow behavior issues

### PR DESCRIPTION
**Pull request by danlim715**
Keep your PR as a Draft until it's ready for Platform review. A PR is ready for Platform review when it has a teammate approval and tests, linting, and settings checks pass CI. See [these tips](https://depo-platform-documentation.scrollhelp.site/developer-docs/vets-api-pr-tips) on how to avoid common delays in getting your PR merged.

## Summary

- *This work is behind a feature toggle (flipper): NO*
- Added some overflow limits to some fields not properly overflowing, as well as fixed some logic issues resulting in improper overflow for some fields. Additionally, removed logic in the hash_converter that would cause list level overflow to trigger if a field level overflow within the list element was triggered. Left a detailed comment in its place for tracking purposes if the removed logic ever needs to be referenced.
- Pension and Burials benefits team to own this change.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/107201
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/114988

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by nanotone — CHANGES_REQUESTED** [review]
Putting a blocker on this PR because it updates the PDF generator in a way that affects overflow rendering for many scenarios outside of the 0969 (see all the fixtures that changed) including a regression in the 0781v2 overflow:

Screenshot 2025-08-06 at 08 56 20

At first glance, this PR's change to HashConverter will need to be reconsidered, because it removes an array overflow check that is necessary. I will follow up either in this PR thread or in Slack.

**Review by balexandr — APPROVED** [review]

**Review by nanotone — DISMISSED** [review]
LGTM!

File: modules/income_and_assets/lib/income_and_assets/helpers.rb [hunk 1]
```diff
@@ -109,8 +80,9 @@ def split_currency_amount_lg(amount, field_lengths = {})
     # @return [String]
     #
     def get_currency_field(arr, neg_i, field_length)
-      value = arr.length >= -neg_i ? arr[neg_i] : 0
-      format("%0#{field_length}d", value.to_i)
+      value = arr.length >= -neg_i ? arr[neg_i] : nil
+      (field_length - value.length).times { value.prepend(' ') } if value
```
[the remark this answers is not in the document]
Comment (stiehlrod) [reply modules/income_and_assets/lib/income_and_assets/helpers.rb:84]: @danlim715 This seems like a valid concern, what are your thoughts?

**Review by stiehlrod — CHANGES_REQUESTED** [review]
@danlim715 This is a whole lot of changes without any additional specs. Some things you might consider testing:
  - Verify that overflow fields are properly mapped in PDF templates
  - Ensure currency formatting changes don't break existing data display
  - Test with various currency amounts and edge cases
  - Confirm that overflow handling doesn't lose important data
  - Validate that PDF rendering handles the new overflow fields correctly
  
  If coverage is over 90% for your module, please add a screenshot. Thanks so much!

File: modules/income_and_assets/lib/income_and_assets/helpers.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (danlim715) [reply modules/income_and_assets/lib/income_and_assets/helpers.rb:84]: @stiehlrod calling `.times` with a negative number does not result in an `ArgumentError`.  If the number is negative, it doesn't run the logic in the block.

**Comment by danlim715** [comment]
> @danlim715 This is a whole lot of changes without any additional specs. Some things you might consider testing:
> 
> * Verify that overflow fields are properly mapped in PDF templates
> * Ensure currency formatting changes don't break existing data display
> * Test with various currency amounts and edge cases
> * Confirm that overflow handling doesn't lose important data
> * Validate that PDF rendering handles the new overflow fields correctly
> 
> If coverage is over 90% for your module, please add a screenshot. Thanks so much!

image
